airflow/kubernetes/pod_launcher.py

```diff
@@ -178,12 +178,17 @@ def monitor_pod(self, pod, get_logs):
         if self.extract_xcom:
             result = self._extract_xcom(pod)
             self.log.info(result)
-            result = json.loads(result)
-        self.istio.handle_istio_proxy(self.read_pod(pod))
+            try:
+                _result = json.loads(result)
+            except:
+                self.log.error(f"Failed to JSON decode xcom: {result}")
         while self.pod_is_running(pod):
+            self.istio.handle_istio_proxy(self.read_pod(pod))
```

Comment:
What was the issue with the other way, where we just set a boolean the first time we shut down to ensure it's only shut-down once?

Also it might make sense to change the name of this function to something like "attempt_graceful_shutdown"

Reply:
I think that way is fine or better

Reply:
Ok I merged our logic and put this part into the istio class' self.shut_down_called